### DISCUSSION TITLE
Seqr local installation [AWS] redirect : 403 Error

### DISCUSSION THREAD

**Discussion by ksarathbabu** [broadinstitute/seqr#4250] [Q&A] [answered]
Hi Everyone,

I have the seqr installed on a local installation using the docker-compose version. It is installed correctly and works fine when used with the machines IP address. But when I added a load balancer to the instance and redirected it from a URL (instead of direct access to the IP address) I get a 403 error.

Here is the log in the container. 

seqr_1             | {"timestamp": "2024-07-19 14:25:46,443", "severity": "WARNING", "message": "Forbidden (Origin checking failed - https://seqr.xyz.com/ does not match any trusted origins.): /api/login"}
seqr_1             | {"timestamp": "2024-07-19 14:25:46,443", "severity": "WARNING", "message": "Forbidden (Origin checking failed - https://seqr.xyz.com/ does not match any trusted origins.): /api/login"}
seqr_1             | {"timestamp": "2024-07-19 14:25:46,443", "severity": "WARNING", "httpRequest": {"requestMethod": "POST", "requestUrl": "http://seqr.xyz.com/api/login", "status": 403, "responseSize": 2584, "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36", "remoteIp": "xyz", "referer": "https://seqr.xyz.com/login/", "protocol": "HTTP/1.1"}, "requestBody": {"email": "[test@test.com](mailto:test@test.com)", "password": "***"}}

Any suggestions on changes I could make to the application so it doesn`t think the URL is the IP address. 

Thank you

**Comment by hanars** [comment]
Try setting the `BASE_URL` environment variable to match your URL. You may also need to set the `DEPLOYMENT_TYPE` environment variable to `prod`

**Reply by ksarathbabu** [reply]
Thank you @hanars for the quick response, unfortunately it didn't help. Am I doing it right? 

#DEPLOYMENT_TYPE = os.environ.get('DEPLOYMENT_TYPE')
#BASE_URL = os.environ.get("BASE_URL", "/")
DEPLOYMENT_TYPE = "prod"
BASE_URL = "https://seqr.xyz.com"

Then I did docker-compose down and docker-compose up.

**Reply by hanars** [reply]
I was suggesting you edit the `environment` section in your docker compose file. I do not know how you are building seqr but the default installation does not support editing source files

**Reply by ksarathbabu** [reply]
Thank you, I added it to the docker-compose seqr environment variable section and still get the same error. 

seqr_1             | {"timestamp": "2024-07-23 15:45:52,629", "severity": "WARNING", "message": "Forbidden (Origin checking failed - https://seqr.xyz.com does not match any trusted origins.): /api/login"}
seqr_1             | {"timestamp": "2024-07-23 15:45:52,629", "severity": "WARNING", "message": "Forbidden (Origin checking failed - https://seqr.xyz.com does not match any trusted origins.): /api/login"}
seqr_1             | {"timestamp": "2024-07-23 15:45:52,629", "severity": "WARNING", "httpRequest": {"requestMethod": "POST", "requestUrl": "http://seqr.xyz.com/api/login", "status": 403, "responseSize": 1019, "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36", "remoteIp": "IPAddress", "referer": "https://seqr.xyz.com/login/", "protocol": "HTTP/1.1"}, "requestBody": {"email": "useremail", "password": "***"}}

**Reply by hanars** [reply]
To clarify, you are seeing this error in google chrome as your browser?

**Reply by ksarathbabu** [reply]
Yes, same error if I use safari as well. 

After googling to solve the problem, tried different solutions like 

CORS_ALLOW_HEADERS = ['*']
CORS_ALLOWED_ORIGINS=['http://seqr.xyz.com','https://seqr.xyz.com']
CSRF_TRUSTED_ORIGINS=['http://seqr.xyz.com','https://seqr.xyz.com']

Unfortunately none worked.

**Reply by hanars** [reply]
you can try setting `DEBUG=True` in settings.py to get a more detailed error log

**Reply by ksarathbabu** [reply, marked as answer]
I wanted to update everyone that once I added CSRF_TRUSTED_ORIGINS = ["https://seqr.xyz.com/"], it was fixed. 

Thanks
